backend/utils/azure_translate.py

```diff
@@ -0,0 +1,74 @@
+import requests
+from urllib.parse import urlencode
+import traceback
+import os
+
+
+class AzureTranslator:
+    def __init__(
+        self,
+        subscription_key: str,
+        region: str,
+        endpoint: str = "https://api.cognitive.microsofttranslator.com",
+    ) -> None:
+        self.http_headers = {
+            "Ocp-Apim-Subscription-Key": subscription_key,
+            "Ocp-Apim-Subscription-Region": region,
+        }
+        self.translate_endpoint = endpoint + "/translate?api-version=3.0&"
+        self.languages_endpoint = endpoint + "/languages?api-version=3.0"
+
+    def get_supported_languages(self) -> dict:
+        return requests.get(self.languages_endpoint).json()["translation"]
+
+    def batch_translate(self, texts: list, src_lang: str, tgt_lang: str) -> list:
+        if not texts:
+            return texts
+
+        if src_lang not in self.supported_languages:
+            raise NotImplementedError(
+                f"Source language code: `{src_lang}` not supported!"
+            )
+
+        if tgt_lang not in self.supported_languages:
+            raise NotImplementedError(
+                f"Target language code: `{tgt_lang}` not supported!"
+            )

```

Comment:
One minor thing : Now that `self.supported_languages` is removed, these lines will throw error.

Reply:
Removed this segment as well.